typescript/0295-find-median-from-data-stream.ts

```diff
@@ -0,0 +1,55 @@
+class MedianFinder {
+    private array: number[];
+
+    constructor() {
+        this.array = [];
+    }
+
+    addNum(num: number): void {
+        if (this.array.length === 0) {
+            this.array.push(num);
+            return;
+        }
+
+        let left = 0;
+        let right = this.array.length - 1;
+
+        while (left <= right) {
+            const mid = Math.floor((left + right) / 2);
+
+            if (this.array[mid] === num) {
+                this.array.splice(mid, 0, num);
```

Comment:
Your time complexity is (N * log(N), if you use a heap, you can lower it to log(N)